packages/three-instanced-uniforms-mesh/src/InstancedUniformsDerivedMaterial.js

```diff
@@ -24,9 +24,14 @@ export function createInstancedUniformsDerivedMaterial (baseMaterial) {
       _uniformNames.forEach((name) => {
         let vertType = vertexUniforms[name]
         let fragType = fragmentUniforms[name]
-        if (vertType || fragType) {
+        const type = vertType || fragType
+        if (type) {
+          // remove existing deklarations
+          const findDeklaration = new RegExp(`\\s*(?:varying|attribute|uniform)\\s+${type}\\s+${name}\\s*;?`, 'g')
+          vertexShader = vertexShader.replaceAll(findDeklaration,"")
```

Comment:
This seems fine to me, but out of curiosity why did you need to remove the existing declaration?

Not critical, but deklaration->declaration would be easier on my English-speaking eyes. 😉

Reply:
I changed it to declaration, and thank you for the correction, my german fingers just seem to like the harder consonants by nature ;)

I need to remove existing declarations for "troika_attr_uvTransform", because otherwise I get an "Vertex shader is not compiled" error because of an "troika_attr_uvTransform' : redefinition", it was allready declared as "uniform mat3 troika_attr_uvTransform;" by threejs and this collides with the new "attribute mat3 troika_attr_uvTransform;":

[shader-error.txt](https://github.com/protectwise/troika/files/7801690/shader-error.txt)

Reply:
Ah, I see! I think that is caused by it renaming the original uniform declaration when it shouldn't -- at the top of the file there's a `precededByUniformRE` regex that should include mat3/mat4 now, I think that would also fix the issue by preventing the declaration from getting renamed.

Although, I'm thinking your approach may be better anyway, since it doesn't leave unused uniforms in the code ...? In which case we could remove the `precededByUniformRE` checks altogether. 🤔 Let me test this out and I may go ahead and remove that after merging your code.